opentasks/src/main/java/org/dmfs/tasks/detailsscreen/TaskDetailsUi.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.tasks.detailsscreen;
+
+import android.content.Context;
+
+
+/**
+ * Represents the task details UI which can be shown.
+ *
+ * @author Gabor Keszthelyi
+ */
+public interface TaskDetailsUi
```

Comment:
The interface looks quite generic. Why not just call it after something that can launch any activity? I mean, how would the same interface for the task editor?

Reply:
You're right, this can be more generic. How about something like this?:

```java
interface Showable<C extends Context>
{
    void show(C context);
}
```

It could represent a 'meaningful' UI of the app that can be shown. By 'meaningful' I mean that it represents a use-case, like 'details of the task', or like a 'permission request dialog'. Something that makes sense on its own and may be fired/shown independently of the place(s) where it is used.

The followings could be covered by this for example:
- opening a new screen by starting an `Activity` (C as Context)
- showing a `DialogFragment`  (C as Activity to call getFragmentManager())
- showing a `Toast` (C as Context)
- showing a `SnackBar` ([using](https://stackoverflow.com/a/8817003/4247460) findViewById(..content) (C as Activity)
- showing a `Notification` (using activity.getSystemService(NotificationManager).notify(n)) (C as Context)

Fragment transactions would not be included. (Except for `DialogFragment`)

Of course we wouldn't need to actually start to use it everywhere, I mentioned them just for the concept.
Let me know if you think that interface makes sense, I would update the code in this PR in that case.